### PR TITLE
update(Files): Add Preview for PDF and Code Files

### LINKED ISSUE #1801: chore(files): oog and aac
STR:
- go to chat
- send .aac or .oog
- the audio from those don't play

macOS, m1

### LINKED ISSUE #1826: bug(files): Investigate why PDF wont open on Windows
**Issue:** 
When you try to open a pdf on Windows, it will just show a blank screen instead of the actual pdf

**OS:**
Windows

**Screen recording:**

https://github.com/Satellite-im/Uplink/assets/93608357/105c1cab-852a-4378-83f9-280372a57a37

### PR DESCRIPTION
**Pull request by lgmarchi**
### 1. Changed Icons on files, now files that are docs, excel, ppt, language codes will have different icons and unknown files. 
image

### 3. Possible to preview PDF files 

https://github.com/Satellite-im/Uplink/assets/63157656/617346ef-4604-46bb-9d9c-e738254ce4b7

### 4. Possible to preview Language Code files 

https://github.com/Satellite-im/Uplink/assets/63157656/bacbd8bd-9f9a-48a2-bf46-c851ca7304db

### 5. Remove ogg from audio files and change audio layout on preview 

Screenshot 2024-02-06 at 11 17 15

### Which issue(s) this PR fixes 🔨

- Resolve #1801 
- Resolve #1826

### CODE REVIEW

**Comment by phillsatellite** [comment]
DM'd Lucas with a small issue I found. I noticed I could preview code language files in File Storage but not in chats. He said he is going to make a new commit for that 👍 

https://github.com/Satellite-im/Uplink/assets/93608357/ebd8d63b-d107-419e-8516-ab9e7589f0e7

File: ui/src/layouts/storage/files_layout/file_preview.rs [hunk 1]
```diff
@@ -129,6 +136,15 @@ fn FilePreview<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
...
 
+    let code_content = if is_code {
+        match fs::read_to_string(file_path_in_local_disk.read().clone()) {
+            Ok(content) => content,
+            Err(_) => String::new(),
+        }
+    } else {
+        String::new()
+    };
+
```
Comment (dariusc93) [inline ui/src/layouts/storage/files_layout/file_preview.rs]: We can simplify this logic to something like the following

```suggestion
let code_content = is_code
        .then(|| std::fs::read_to_string(file_path_in_local_disk.read().clone()).ok())
        .flatten()
        .unwrap_or_default();

```

File: ui/src/layouts/storage/files_layout/file_preview.rs [hunk 2]
```diff
@@ -208,5 +235,57 @@ fn FileTypeTag(cx: Scope<FileTypeTagProps>) -> Element {
...
+                }
+            ),
+    },})
+}
+
+fn get_language_class(file_path: &str) -> String {
+    let extension = Path::new(file_path)
+        .extension()
+        .and_then(std::ffi::OsStr::to_str)
+        .unwrap_or("");
```
Comment (dariusc93) [inline ui/src/layouts/storage/files_layout/file_preview.rs]:
```suggestion
        .unwrap_or_default();
```

**Comment by phillsatellite** [comment]
@lgmarchi Just one other thing (maybe we can get another Windows user to test this out) but when I open a pdf on Mac/Linux it opens fine with no issues but on Windows the file will open but with no content inside of preview. I tried to scroll up in Windows to see if maybe the content was just out of view but no luck 

MacOS: 
image

Windows: 
image

**Comment by lgmarchi** [comment]
> @lgmarchi Just one other thing (maybe we can get another Windows user to test this out) but when I open a pdf on Mac/Linux it opens fine with no issues but on Windows the file will open but with no content inside of preview. I tried to scroll up in Windows to see if maybe the content was just out of view but no luck
> 
> MacOS: image
> 
> Windows: image

Interesting and very weird my friend. I will investigate.

**Review by phillsatellite — APPROVED** [review]

File: common/src/lib.rs [hunk 3]
```diff
@@ -266,6 +275,20 @@ pub fn return_correct_icon(file_name: &str) -> Icon {
...
+    if is_video(file_name) {
+        FileType::Video
+    } else if is_audio(file_name) {
+        FileType::Audio
+    } else if is_doc(file_name) {
+        FileType::Doc
+    } else if is_lang_file(file_name) {
+        FileType::Code
+    } else {
+        FileType::Image
```
Comment (dariusc93) [inline common/src/lib.rs:307]: What if a file is not an image? Would it be better to have a generic type that would be the default?

Comment (lgmarchi) [reply common/src/lib.rs:307]: Changed!

**Review by dariusc93 — APPROVED** [review]

**Review by stavares843 — COMMENTED** [review]
appears this icon when upload .gif for example

Captura de ecrã 2024-02-08, às 15 27 32

macOS, m1

**Review by stavares843 — CHANGES_REQUESTED** [review]
when clicking on a image that was sent appears this


https://github.com/Satellite-im/Uplink/assets/29093946/1451e59f-f471-48f5-b7aa-a7e4d3aaea8a


macOS, m1

**Comment by phillsatellite** [comment]
When testing after the latest commits, Ive also noticed that you have to click on a file a couple times to get the preview to open. On first attempt it will say "it is not possible to preview this file" then if you click it again after that file preview will open 

https://github.com/Satellite-im/Uplink/assets/93608357/8652de30-f19f-4821-9b0c-656f377d6cd5

**Comment by stavares843** [comment]
also i can open pdf's i sent fine but if i try to open any pdf's phil sent me does not open and no error appears

https://github.com/Satellite-im/Uplink/assets/29093946/ad0b7bf8-1544-4500-a55d-a107bbd6dbd2

Captura de ecrã 2024-02-08, às 15 33 28

**Review by stavares843 — APPROVED** [review]

**Review by stavares843 — APPROVED** [review]

**Review by phillsatellite — APPROVED** [review]

**Pull request merged**